docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,121 @@
+2. Analyze and process on-premises data (by use of visualizations, dashboards, etc.) at Red Hat like SaaS data.
+
+
+### Non-Goals
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+
+
+## Proposal
+
+A new flow for sending data is to be added:
+When a cluster installation attempt finishes (regardless of success or failure), all of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query data
+- Needs to know how to format the data to send to the data hosting service
+- Needs to expose an option for users to opt out of this feature
+
+Additional changes needed:
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment
+
+Sending data should not interfere with customer workloads and deployment
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do additional configurations. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
```

Comment:
Do we want to add a retry mechanism for sending the data in case the DATA_COLLECTION_URL is not available?

Reply:
That's a good idea! Perhaps the retry mechanism could be to retry X times after Y minutes has passed.
E.g. Retry sending 5 more times every 10 minutes until success or exceeded number of retries.

What do you think?

Reply:
suggesting to start with best effort first, retry can be an improvement and it cannot be a blocking operation.
meaning if we failed to send any data we cannot block the operation or the service because of it. 
As Crystal mentioned in `Risk 1: Overusing Customer's Network & CPU`

Reply:
Agreed! Modified the doc to include this. 

> retry can be an improvement

Do you mean retry can be implemented later and not as part of this first pass at the design?

Reply:
implemented later

Reply:
Gotcha, thanks! Modified the doc to mention retry as a future enhancement

Reply:
Per the 3rd customer story: Are you discussing existing events or new ones with extra data? Because for our ELK the data in the event is processed first in order to provide a meaningful insight.

Reply:
I think it would include both existing and new events (all data from SaaS and the on-premises deployments)